api/src/main/java/jakarta/servlet/ServletOutputStream.java

```diff
@@ -304,4 +362,16 @@ public void println(double d) throws IOException {
      */
     public abstract void setWriteListener(WriteListener writeListener);
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * If this method is called when the output stream is in non-blocking mode it will immediately return even if the stream
+     * contains buffered data that is yet to be written to client. The container will write this data out in the background,
+     * and then close the response. If this process fails the {@link WriteListener#onError(Throwable)} method will be
+     * invoked as normal.
```

Comment:
Is it "and then close the response"?    I think the response is functionally closed when the method returns (write cannot be called again), but that the return does not guarantee that the data has been written.
```suggestion
     * If this method is called when the output stream is in non-blocking mode it will immediately return with the stream effective close, even if the stream
     * contains buffered data that is yet to be written to client. The container will write this data out in the background. If this process fails the {@link WriteListener#onError(Throwable)} method will be
     * invoked as normal.
```

Reply:
`with the stream effective close` should be `with the stream effectively closed`